scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/ManageLabelsWindow.java

```diff
@@ -0,0 +1,214 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.scheduler.client;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.ow2.proactive_grid_cloud_portal.common.client.Images;
+
+import com.smartgwt.client.types.Alignment;
+import com.smartgwt.client.types.ListGridFieldType;
+import com.smartgwt.client.widgets.IButton;
+import com.smartgwt.client.widgets.Window;
+import com.smartgwt.client.widgets.form.validator.CustomValidator;
+import com.smartgwt.client.widgets.form.validator.Validator;
+import com.smartgwt.client.widgets.grid.ListGrid;
+import com.smartgwt.client.widgets.grid.ListGridField;
+import com.smartgwt.client.widgets.grid.ListGridRecord;
+import com.smartgwt.client.widgets.layout.HLayout;
+import com.smartgwt.client.widgets.layout.VLayout;
+
+
+/**
+ * Client-side Manage labels menu
+ *
+ * Changes or newly added records are made in memory and not persisted
+ *
+ */
+public class ManageLabelsWindow {
+
+    public static final String LABEL_REGEX = "^[a-zA-Z0-9_/-]*$";
+
+    public static final int LABEL_MAX_LENGTH = 20;
+
+    private Window window;
+
+    private final SchedulerController controller;
+
+    private ListGrid labelsListGrid;
+
+    public ManageLabelsWindow(SchedulerController controller) {
+        this.controller = controller;
+        this.build();
+    }
+
+    public void show() {
+        this.window.show();
+    }
+
+    public void destroy() {
+        this.window.destroy();
+    }
+
+    private void build() {
+
+        labelsListGrid = new ListGrid();
+        labelsListGrid.setWidth(440);
+        labelsListGrid.setHeight(150);
+        labelsListGrid.setAlternateRecordStyles(true);
+        labelsListGrid.setShowAllRecords(true);
+
+        ListGridField labelField = new ListGridField("label", "Label", 390);
+        Validator requiredValidator = new CustomValidator() {
+            @Override
+            protected boolean condition(Object value) {
+                return value != null && !value.toString().trim().isEmpty();
+            }
+        };
+        requiredValidator.setErrorMessage("Field is required");
+        Validator lengthValidator = new CustomValidator() {
+            @Override
+            protected boolean condition(Object value) {
+                return value == null || value.toString().length() < LABEL_MAX_LENGTH;
+            }
+        };
+        lengthValidator.setErrorMessage("The maximum length is 20 characters");
```

Comment:
unfortunately, this check and error message does not take into consideration the configurable aspect on the server side

Reply:
From there there are 2 possibilities:
 - you could fetch the properties using rest/scheduler/properties
 - you could perform the validation once the changes are submitted and checking for server side errors

Reply:
should we make it configurable here too or should we remove this validators and handle the backend exception?

Reply:
we should not make it configurable on the portal directly. handling the backend exception is probably the best to do.